### DISCUSSION TITLE
creating-a-source-generator-part-4-customising-generated-code-with-marker-attributes/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#375] [Announcements]
# creating-a-source-generator-part-4-customising-generated-code-with-marker-attributes/

In this post I describe how to customise a source generator by reading properties from a marker attribute

https://andrewlock.net/creating-a-source-generator-part-4-customising-generated-code-with-marker-attributes/

**Comment by ghostnguyen** [comment]
We can create an instance object of attribute to access properties:

(T)Activator.CreateInstance(typeof(T), attributeData.GetActualConstuctorParams().ToArray())

**Reply by andrewlock** [reply]
You can, but you almost certainly shouldn't for performance reasons! 😁

**Reply by ghostnguyen** [reply]
Since this is compile time, it has no effect on performance at run time.

**Reply by andrewlock** [reply]
Correct, it has an effect on IDE performance

**Comment by PieterjanDeClippel** [comment]
Hi, what is `enumAttribute`? It's defined nowhere in this article?
